source/DataAccess/DADGenerator.ts

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import IntegrationConfigFactory from '../IntegrationConfigFactory';
+import { IIntegrationConfig, IntegrationType } from '../IIntegrationConfig';
+import { writeFile } from 'fs';
+
+export default class DataAccessDoc {
+
+    private _integrationType;
+    private _queries;
+    private _header: string[] = [];
+    private _footer: string[] = [];
+
+    constructor(integrationType: IntegrationType) {
+        this._integrationType = integrationType.toString();
+        const icf = new IntegrationConfigFactory();
+        const cfg: IIntegrationConfig = icf.create(this._integrationType);
+        this._queries = cfg.queries;
+        this._header.push(`# Data Access Requirements: ${this._integrationType}`);
+        this._header.push(`DIA requires access to the following tables/fields in the ${this._integrationType} integration type.`);
```

Comment:
Semantically, the last two lines here seem like they fit more cohesively in the `create()` method below, as it is generating actual markdown content.  The preceding lines in the constructor set up class state only.